hawkular-command-gateway-itest/src/test/java/org/hawkular/cmdgw/ws/test/EchoCommandITest.java

```diff
@@ -16,25 +16,125 @@
  */
 package org.hawkular.cmdgw.ws.test;
 
+import org.hamcrest.CoreMatchers;
+import org.hawkular.cmdgw.api.EchoRequest;
+import org.hawkular.cmdgw.api.WelcomeResponse;
+import org.hawkular.cmdgw.ws.test.TestWebSocketClient.ExpectedEvent;
+import org.hawkular.cmdgw.ws.test.TestWebSocketClient.ExpectedEvent.ExpectedClose;
+import org.hawkular.cmdgw.ws.test.TestWebSocketClient.ExpectedEvent.ExpectedFailure;
+import org.hawkular.cmdgw.ws.test.TestWebSocketClient.ExpectedEvent.ExpectedMessage;
+import org.hawkular.cmdgw.ws.test.TestWebSocketClient.MessageAnswer;
+import org.hawkular.cmdgw.ws.test.TestWebSocketClient.PatternMatcher;
 import org.testng.annotations.Test;
 
+import com.squareup.okhttp.Credentials;
+import com.squareup.okhttp.ws.WebSocket;
+
 /**
  * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
  */
 
 public class EchoCommandITest extends AbstractCommandITest {
+    public static final long serverReauthPeriodSeconds =
+            Integer.parseInt(System.getProperty("org.hawkular.cmdgw.reauthIntervalSeconds"));
     public static final String GROUP = "EchoCommandITest";
 
-    @Test(groups = {GROUP})
+    private static final String echoRequestTemplate = "EchoRequest={\"authentication\": " + authentication
+            + ", \"echoMessage\": \"%s\"}";
+    private static final String echoResponseTemplate = "EchoResponse={\"reply\":\"ECHO [%s]\"}";
+
+    /**
+     * Tests the simplest echo request-response exchange.
+     *
+     * @throws Throwable
+     */
+    @Test(groups = { GROUP })
     public void testEcho() throws Throwable {
 
-        String req = "EchoRequest={\"authentication\": " + authentication
-                + ", \"echoMessage\": \"Yodel Ay EEE Oooo\"}";
-        String response = "EchoResponse={\"reply\":\"ECHO [Yodel Ay EEE Oooo]\"}";
-       try (TestWebSocketClient testClient = TestWebSocketClient.builder()
+        try (TestWebSocketClient testClient = TestWebSocketClient.builder()
+                .url(baseGwUri + "/ui/ws") //
+                .expectWelcome(String.format(echoRequestTemplate, "Yodel Ay EEE Oooo")) //
+                .expectText(String.format(echoResponseTemplate, "Yodel Ay EEE Oooo")) //
+                .build()) {
+            testClient.validate(10000);
```

Comment:
> Does it block for 10s, or will it finish before if the message has been received in, say, 1s?

10s is just the timeout for the collecting of conversation reports. If the reports are available earlier, then they are validated immediately.